pkg/volume/aws_ebs/nvme/nvme_linux.go

```diff
@@ -0,0 +1,108 @@
+// +build linux,amd64
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nvme
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	cmd_NVME_ID_CNS_NS   = 0x00
+	cmd_NVME_ID_CNS_CTRL = 0x01
+)
+
+const nvme_admin_identify = 0x06
+const (
+	cmd_NVME_IOCTL_ID        = uintptr(0x40)
+	cmd_NVME_IOCTL_ADMIN_CMD = uintptr(0x41)
+)
+
+const (
+	ioctlmode_IOC_NONE  = 0
+	ioctlmode_IOC_READ  = 2
+	ioctlmode_IOC_WRITE = 1
+)
+
+// Identify returns the nvme volume identifier for an nvme device
+// Based on the calls made by "nvme list"
+func Identify(p string) (string, bool, error) {
```

Comment:
if aws doc about NVMe path is correct (i.e. n1), why not just use ns 1 (without having to probe ns)?